admin/model_search_index_create_request.go

```diff
@@ -5,11 +5,11 @@ package admin
 // SearchIndexCreateRequest struct for SearchIndexCreateRequest
 type SearchIndexCreateRequest struct {
 	// Label that identifies the collection to create an Atlas Search index in.
-	CollectionName string `json:"collectionName"`
+	CollectionName *string `json:"collectionName,omitempty"`
 	// Label that identifies the database that contains the collection to create an Atlas Search index in.
-	Database string `json:"database"`
+	Database *string `json:"database,omitempty"`
 	// Label that identifies this index. Within each namespace, names of all indexes in the namespace must be unique.
-	Name string `json:"name"`
+	Name *string `json:"name,omitempty"`
 	// Type of the index. The default type is search.
```

Comment:
From my understanding this is removing required from all properties present in oneOf schemas. 
This does fix issue in CLOUDP-307850 but is also impacting broader cases were this transformation from required to optional is not needed. 
Selected this specific case of search index, `collectionName` `database` and `name` are required properties in all the oneOf schemas (vector and text search) so we should have no need of marking it as optional.

Would be great if we can dedicate some more time to get a more specific fix.

Reply:
Agree. We can make logic to only remove it when not all types have it required etc.